tools/common_lib/src/gemm.h

```diff
@@ -424,6 +586,69 @@ class Gemm : public DirectMlBaseNode
         record_execute_impl(dml_cmd_recorder, cmd_list, input_bindings, output_bindings);
     }
 
+    virtual void record_initialize(IDMLCommandRecorder* dml_cmd_recorder, ID3D12GraphicsCommandList* cmd_list, ID3D12Resource* resource_b, ID3D12Resource* resource_c,
```

Comment:
just keep single `record_initialize` and pass nullptr for resources which are not used

Reply:
Have restructured the code and made Quant Gemm to be a separate class.